persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java

```diff
@@ -68,228 +72,257 @@
+               MediaType serverValueStorageType = pingResponse.getValueMediaType();
+
+               DataFormat.Builder dataFormatBuilder = DataFormat.builder().from(remoteCache.getDataFormat());
+               Integer numSegments = remoteCache.getCacheTopologyInfo().getNumSegments();
+               boolean segmentsMatch;
+               if (numSegments == null) {
+                  log.debugf("Remote Store for cache %s cannot support segmentation as the number of segments was not found from the remote cache", cacheName);
+                  segmentsMatch = false;
+               } else {
+                  segmentsMatch = numSegments == segmentCount;
+                  if (segmentsMatch) {
+                     log.debugf("Remote Store for cache %s can support segmentation as the number of segments matched the remote cache", cacheName);
+                  } else {
+                     log.debugf("Remote Store for cache %s cannot support segmentation as the number of segments %d do not match the remote cache %d",
+                           cacheName, segmentCount, numSegments);
+                  }
+               }
+               if (!segmentsMatch && configuration.segmented()) {
+                  throw log.segmentationRequiresEqualSegments(segmentCount, numSegments);
+               }
+               if ((supportsSegmentation = localKeyStorageType.equals(serverKeyStorageType))) {
+                  if (localKeyStorageType.isBinary()) {
+                     dataFormatBuilder.keyType(localKeyStorageType);
+                     // Make sure when the media types match that we don't marshall the value
+                     dataFormatBuilder.keyMarshaller(IdentityMarshaller.INSTANCE);
+                  } else {
+                     dataFormatBuilder.keyType(marshaller.mediaType());
+                     dataFormatBuilder.keyMarshaller(marshaller);
+                  }
+               } else if (configuration.segmented()) {
+                  throw log.segmentationRequiresEqualMediaTypes(localKeyStorageType, serverKeyStorageType);
+               }
+               // We store MarshalledEntry instances for the values when raw values is not set, instead of the stored value
+               if (!configuration.rawValues()) {
+                  localValueStorageType = MediaType.APPLICATION_OBJECT;
+               }
+               if (localValueStorageType.equals(serverValueStorageType)) {
+                  if (localValueStorageType.isBinary()) {
+                     dataFormatBuilder.valueType(localValueStorageType);
+                     dataFormatBuilder.valueMarshaller(IdentityMarshaller.INSTANCE);
+                  } else {
+                     dataFormatBuilder.valueType(marshaller.mediaType());
+                     dataFormatBuilder.valueMarshaller(marshaller);
+                  }
+               }
+               DataFormat dataFormat = dataFormatBuilder.build();
+               log.fatalf("Data format is %s", dataFormat);
```

Comment:
I think this needs to be `debugf` :slightly_smiling_face:

Reply:
Yea, this was only here for testing purposes. Let me change this, I might just remove.